docs/reference/projectyml.rst

```diff
@@ -65,29 +65,42 @@ This section explains how to configure a project-wide pip index. This index will
 modules' dependencies.
 By default, a project created using the :ref:`project-creation-guide` is configured to install packages from ``https://pypi.org/simple/``.
 The :class:`~inmanta.module.ProjectPipConfig` section of the project.yml file offers options to configure this behaviour.
+Some of these options are detailed below:
 
 pip.use-system-config
 """""""""""""""""""""
 
-This option determines the isolation level of the project's pip config. When false, any pip config set on the system
+This option determines the isolation level of the project's pip config. When false (the default), any pip config set on the system
 (e.g. through environment variables or pip config files) are ignored and pip will only look for packages
-in the index(es) defined in the project.yml, when true, pip will in addition look in the eventual index(es) defined on the system.
+in the index(es) defined in the project.yml, when true, the orchestrator will extend the system's pip configuration rather than replace it.
 
-Setting this to ``false`` is recommended during development both for portability (by making sure that only the pip
-config defined in the project.yml will be used regardless of the sytem's pip config) and for security (The isolation
-reduces the risk of dependency confusion attacks if the ``index-url`` option is set mindfully).
+Setting this to ``false`` is generally recommended, especially during development, both for portability (achieving
+consistent behavior regardless of the system it runs on, which is important for reproductive testing on developer
+machines, easy compatibility with Inmanta pytest extensions, and consistency between compiler and agents) and for
+security (the isolation reduces the risk of dependency confusion attacks).
 
 Setting this to ``true`` will have the following consequences:
 
 - If no index is set in the project.yml file i.e. both ``index-url`` and ``extra-index-url`` are unset, then Pip's
   default search behaviour will be used: environment variables, pip config files and then PyPi (in that order).
 
-- If ``index-url`` and/or ``extra-index-url`` are set, they will be used and any index defined in the system's environment
-  variables or pip config files will also be used (in that order) and passed to pip as extra indexes.
+- If ``index-url`` is set, this value be used over any index defined in the system's environment
+  variables or pip config files.
 
-- The ``PIP_PRE`` environment variable (if set) is no longer ignored and will be used to determine whether pre-release
+- If ``extra-index-url`` is set, these indexes will be used in addition to any extra index defined in the system's
+  environment variables or pip config files, and passed to pip as extra indexes.
+
+- If ``PIP_PRE`` is set, pre-release
```

Comment:
For the other bullet points you focus on the options in our config file and how they override (or don't) the system config. Here you suddenly take a different tack.